rust/lance-table/src/io/commit.rs

```diff
@@ -95,45 +179,67 @@ async fn current_manifest_path(
         }
     }
 
-    // We use `list_with_delimiter` to avoid listing the contents of child directories.
-    let manifest_files = object_store
-        .inner
-        .list_with_delimiter(Some(&base.child(VERSIONS_DIR)))
-        .await?;
+    let manifest_files = object_store.inner.list(Some(&base.child(VERSIONS_DIR)));
 
-    let current = manifest_files
-        .objects
-        .into_iter()
-        .filter(|meta| {
-            meta.location.filename().is_some()
-                && meta
-                    .location
-                    .filename()
-                    .unwrap()
-                    .ends_with(MANIFEST_EXTENSION)
-        })
-        .filter_map(|meta| {
-            let version = meta
-                .location
-                .filename()
-                .unwrap()
-                .split_once('.')
-                .and_then(|(version_str, _)| version_str.parse::<u64>().ok())?;
-            Some((version, meta))
-        })
-        .max_by_key(|(version, _)| *version);
-
-    if let Some((version, meta)) = current {
-        Ok(ManifestLocation {
-            version,
-            path: meta.location,
-            size: Some(meta.size as u64),
-        })
-    } else {
-        Err(Error::NotFound {
-            uri: manifest_path(base, 1).to_string(),
+    let mut valid_manifests = manifest_files.try_filter_map(|res| {
+        if let Some(scheme) = ManifestNamingScheme::detect_scheme(res.location.filename().unwrap())
+        {
+            future::ready(Ok(Some((scheme, res))))
+        } else {
+            future::ready(Ok(None))
+        }
+    });
+
+    let first = valid_manifests.next().await.transpose()?;
```

Comment:
So in the case of v2 we only look at the first result.  Is there value in some kind of `debug_assert` that looks at the remaining asserts and ensures our understanding of object stores is correct?  E.g. what happens if some new object store (e.g. r2, digital ocean, etc.) decides to use a different convention?  Would we catch it?

Reply:
Added a sanity check that looks at the first 1k results to see if they are ordered. This is the page size in object store list operations. I've verified there is no latency impact on S3.